lib/mini_sql/mysql/deserializer_cache.rb

```diff
@@ -15,13 +15,11 @@ def materialize(result, decorator_module = nil)
         key = result.fields
 
         # trivial fast LRU implementation
-        materializer = @cache.delete(key)
-        if materializer
-          @cache[key] = materializer
-        else
-          materializer = @cache[key] = new_row_matrializer(result)
-          @cache.shift if @cache.length > @max_size
-        end
+        materializer =
```

Comment:
This is no longer an LRU, stuff will be bumped out of the cache even if it was used a micro second ago.

"delete" is critical here, otherwise the hash ordering is not corrected to account for recent usage.

Reply:
I agree